x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/querydsl/query/SpatialRelatesQuery.java

```diff
@@ -228,10 +227,10 @@ private static org.apache.lucene.search.Query pointShapeQuery(
             if (geometry == null || geometry.isEmpty()) {
                 throw new QueryShardException(context, "Invalid/empty geometry");
             }
-            if (geometry.type() != ShapeType.POINT && relation == ShapeField.QueryRelation.CONTAINS) {
+            final XYGeometry[] luceneGeometries = LuceneGeometriesUtils.toXYGeometry(geometry, t -> {});
+            if (isPointGeometry(luceneGeometries) == false && relation == ShapeField.QueryRelation.CONTAINS) {
```

Comment:
I imagine an edge case where we have a multi-point, but all points are the same, or at least quantize to the same point.

Reply:
indeed, this requires a bit more work but this is still an improvement.

Reply:
I see this fix was already in the geo_point case from years ago. So this increases symmetry too.